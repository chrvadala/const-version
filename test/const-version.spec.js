/* global describe, it, expect, beforeEach, afterAll, test */
const mock = require('mock-fs')
const constVersion = require('../src/const-version')
const fs = require('fs')

beforeEach(() => {
  mock({
    dummy_fs: {
      good: {
        'package.json': '{"version": "1.2.3",  "name" : "abc"}'
      },
      missing_version: {
        'package.json': '{"name" : "abc"}'
      },
      malformed_json: {
        'package.json': '{"version": "1.2.3"'
      }
    }
  })
})

afterAll(() => {
  mock.restore()
})

describe('const version', () => {
  it('should read package.json and save file with version', async () => {
    const source = 'dummy_fs/good/package.json'
    const dest = 'dummy_fs/ver.js'
    await constVersion(source, dest)
    const generatedFile = fs.readFileSync(dest, 'utf-8')
    expect(generatedFile).toMatch('1.2.3')
    mock.restore()
    expect(generatedFile).toMatchSnapshot()
  })

  test.each([
    { source: 'dummy_fs/malformed_json/package.json', expectedError: "file doesn't contain valid json" },
    { source: 'dummy_fs/notfound_package.json', expectedError: 'file not found' },
    { source: 'dummy_fs/missing_version/package.json', expectedError: "file doesn't contain version" }
  ])('should be able to handle failing scenarious. Error: \'$expectedError\'', async ({ source, expectedError }) => {
    const dest = 'dummy_fs/ver.js'
    await expect(constVersion(source, dest)).rejects.toThrow(expectedError)
    expect(fs.existsSync(dest)).toBe(false)
  })
})
