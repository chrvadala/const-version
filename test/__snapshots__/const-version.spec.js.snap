// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`const version should read package.json and save file with version 1`] = `
"// THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
export const VERSION = '1.2.3';"
`;
